Libs/RaptureWebServlet/src/main/java/rapture/server/web/servlet/UserBoundedScriptPageServlet.java

```diff
@@ -0,0 +1,218 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package rapture.server.web.servlet;
+
+import java.io.IOException;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpStatus;
+import org.apache.log4j.Logger;
+
+import rapture.common.CallingContext;
+import rapture.common.ErrorWrapper;
+import rapture.common.RaptureScript;
+import rapture.common.RaptureURI;
+import rapture.common.exception.ExceptionToString;
+import rapture.common.exception.RaptureException;
+import rapture.common.exception.RaptureExceptionFactory;
+import rapture.common.impl.jackson.JacksonUtil;
+import rapture.common.model.ContextResponseData;
+import rapture.common.model.RaptureUser;
+import rapture.kernel.ContextFactory;
+import rapture.kernel.Kernel;
+import rapture.kernel.script.KernelScript;
+import reflex.ReflexException;
+import reflex.ReflexExecutor;
+
+/**
+ * The User bounded servlet runs Reflex scripts with (or 'bound' to) a specified user.
+ * 
+ * This servlet take two web.xml params:
+ * 1. user    An existing Rapture user setup with appropriate entitlements 
+ * 2. prefix  A defined location where only this servlet can run scripts
+ * 
+ * @author jonathan-major
+ *
+ */
+public class UserBoundedScriptPageServlet extends BaseServlet {
+
+    /**
+     * 
+     */
+    private static CallingContext DEFINEDUSER = null;
+    private static final long serialVersionUID = 1L;
+    private static final Logger log = Logger.getLogger(UserBoundedScriptPageServlet.class);
+
+    private String scriptPrefix;
+    private String user;
+     
+
+    @Override
+    public void init() throws ServletException {
+        log.info("Starting UserBoundedScriptPageServlet");
```

Comment:
If you're changing this file (for the other things), you should change these to debug. It's not that interesting except when you're debugging.